lit_gpt/utils.py

```diff
@@ -395,6 +395,15 @@ def CLI(*args: Any, **kwargs: Any) -> Any:
     return CLI(*args, **kwargs)
 
 
+def save_hyperparameters(function: callable, checkpoint_dir: Path) -> None:
+    """Captures the CLI parameters passed to `function` without running `function` and saves them to the checkpoint."""
+    from jsonargparse import capture_parser
+
+    parser = capture_parser(lambda: CLI(function))
+    config = parser.parse_args()
+    parser.save(config, checkpoint_dir / "hyperparameters.yaml", overwrite=True)
```

Comment:
I don't like that we have a "config_hub" but this generates "hyperparameters". We should stick to one name for these yamls

Reply:
I don't understand. What is the name you want concretely?
This is for reproducibility, the file should be a fixed deterministic name. How am I going to load it otherwise in generate.py/chat.py?

Reply:
To be clear, this is not meant to be for the config hub. But it's nice if you've trained many experiments and then after evaluation, you can just pick a checkpoint and copy out the hyperparameters.yaml into the config_hub if you like. Without you having to remember what the input args were. I mean, there is no way to know otherwise if we don't save this.